src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+                tooltip.accept(Helpers.translatable("tfc.composter.rotten"));
+            }
+            else
+            {
+                hoeOverlay(level, block, entity, tooltip);
+                if (!composter.isReady())
+                {
+                    final String percent = Helpers.formatPercentage(Math.min(0.99f, (float) composter.getTicksSinceUpdate() / composter.getReadyTicks()) * 100);
+                    tooltip.accept(Helpers.translatable("tfc.jade.progress").append(percent));
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip CROP = (level, state, entity, tooltip) -> {
+        if (entity instanceof CropBlockEntity crop && state.getBlock() instanceof CropBlock block)
+        {
+            hoeOverlay(level, block, entity, tooltip);
+
+            tooltip.accept(Helpers.translatable("tfc.jade.yield").append(Helpers.literal(String.format("%.2f", crop.getYield()))));
+        }
+    };
+
+    public static final BlockEntityTooltip CRUCIBLE = (level, state, entity, tooltip) -> {
+        if (entity instanceof CrucibleBlockEntity crucible)
+        {
+            crucible.getCapability(HeatCapability.BLOCK_CAPABILITY).ifPresent(cap -> heat(tooltip, cap.getTemperature()));
+        }
+    };
+
+    public static final BlockEntityTooltip FIREPIT = (level, state, entity, tooltip) -> {
+        if (entity instanceof AbstractFirepitBlockEntity<?> firepit)
+        {
+            heat(tooltip, firepit.getTemperature());
+
+            if (firepit instanceof PotBlockEntity pot)
+            {
+                if (pot.shouldRenderAsBoiling())
+                {
+                    tooltip.accept(Helpers.translatable("tfc.tooltip.pot_boiling"));
+                }
+                else if (pot.getOutput() != null && !pot.getOutput().isEmpty())
+                {
+                    tooltip.accept(Helpers.translatable("tfc.tooltip.pot_finished"));
+
+                    if (pot.getOutput() instanceof SoupPotRecipe.SoupOutput soup)
+                    {
+                        final ItemStack stack = soup.stack();
+                        itemWithCount(tooltip, stack);
+
+                        final List<Component> text = new ArrayList<>();
+                        stack.getCapability(FoodCapability.CAPABILITY).ifPresent(cap -> cap.addTooltipInfo(stack, text));
+                        text.forEach(tooltip);
+                    }
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip FRUIT_TREE_SAPLING = (level, state, entity, tooltip) -> {
+        if (entity instanceof TickCounterBlockEntity counter && state.getBlock() instanceof FruitTreeSaplingBlock sapling)
+        {
+            final String growth = Helpers.formatPercentage(Math.min(0.99f, (float) counter.getTicksSinceUpdate() / (sapling.getTreeGrowthDays() * ICalendar.TICKS_IN_DAY)) * 100);
+            tooltip.accept(Helpers.translatable("tfc.jade.growth").append(growth));
```

Comment:
Honestly, I think this tooltip as a percentage is more misleading than it would be helpful, because growth time does not advance linearly, it advances based on random tick. What might be more useful is some sort of "Growing" / "Cannot Grow" tooltip, but I think the hoe overlay below already covers that (in which case, this whole entry might be removed in favor of just using `HOE_OVERLAY`